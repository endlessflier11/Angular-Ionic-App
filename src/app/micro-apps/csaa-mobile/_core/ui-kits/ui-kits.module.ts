import { NgModule } from '@angular/core';
import { CommonModule, DatePipe } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { IonicModule } from '@ionic/angular';
import { ThemeIconComponent } from './theme-icon/theme-icon.component';
import { WorkingHoursComponent } from './working-hours/working-hours.component';
import { MyPolicyCardComponent } from './my-policy-card/my-policy-card.component';
import { SkeletonCardComponent } from './skeleton-card/skeleton-card.component';
import { BackButtonComponent } from './back-button/back-button.component';
import { PhonePipe } from './pipes/phone-number.pipe';
import { CsaaAutoIdCardComponent } from './csaa-auto-id-card/csaa-auto-id-card.component';
import { CsaaLoadingComponent } from './csaa-loading/csaa-loading.component';
import { ContentRendererComponent } from './content-renderer/content-renderer.component';
import { NationalCustomAlertModalComponent } from './national-custom-alert/national-custom-alert-modal';
import { SafeDataPipe } from './pipes/safe-data/safe-data.pipe';
import { DocumentActionModalComponent } from './document-action-modal/document-action-modal.component';
import { CsaaPolicySubtitlesComponent } from './csaa-policy-subtitles/csaa-policy-subtitles.component';
import { PaperlessTermsComponent } from './paperless-terms/paperless-terms.component';
import { PaperlessTermsModalComponent } from './paperless-terms-modal/paperless-terms-modal.component';
import { AppleWalletButtonComponent } from './apple-wallet-button/apple-wallet-button.component';

const UI_COMPONENTS = [
  ThemeIconComponent,
  WorkingHoursComponent,
  MyPolicyCardComponent,
  SkeletonCardComponent,
  BackButtonComponent,
  CsaaAutoIdCardComponent,
  CsaaLoadingComponent,
  ContentRendererComponent,
  NationalCustomAlertModalComponent,
  DocumentActionModalComponent,
  CsaaPolicySubtitlesComponent,
  PaperlessTermsComponent,
  PaperlessTermsModalComponent,
  AppleWalletButtonComponent,
];

const UI_PIPES = [PhonePipe, SafeDataPipe];

@NgModule({
  imports: [CommonModule, ReactiveFormsModule, IonicModule],
  declarations: [...UI_COMPONENTS, ...UI_PIPES],
  exports: [...UI_COMPONENTS, ...UI_PIPES],
  providers: [DatePipe],
})
export class UiKitsModule {}
