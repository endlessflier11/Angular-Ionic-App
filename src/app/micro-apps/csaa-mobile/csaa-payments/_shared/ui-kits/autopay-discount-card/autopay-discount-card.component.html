<ion-item
  detail
  lines="none"
  (click)="clickHandler()"
  [detailIcon]="navigate ? 'chevron-forward' : isOpen ? 'chevron-up' : 'chevron-down'"
  [class.ion-no-padding]="navigate"
>
  <ion-label>
    <h3>Save money with AutoPay</h3>
    <ion-text color="dark">
      <p class="ion-text-wrap">
        Save up to {{ discountDataDiff | currency: 'USD':'symbol':'1.0-0' }} per payment with
        <br />AutoPay. Tap to view fees
      </p>
    </ion-text>
  </ion-label>
</ion-item>
<div class="discount" [class.active]="isOpen">
  <table>
    <thead>
      <tr>
        <th>
          <h3>Auto Payment Method Installment Fees</h3>
        </th>
        <th></th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Checking or Savings Account</td>
        <td>
          {{ payment?.autopayInstallmentFee?.autoPayFees?.eft | currency: 'USD':'symbol':'1.2-2' }}
        </td>
      </tr>
      <tr>
        <td>Debit Card</td>
        <td>
          {{
            payment?.autopayInstallmentFee?.autoPayFees?.pciDebitCard
              | currency: 'USD':'symbol':'1.2-2'
          }}
        </td>
      </tr>
      <tr>
        <td>Credit Card</td>
        <td>
          {{
            payment?.autopayInstallmentFee?.autoPayFees?.pciCreditCard
              | currency: 'USD':'symbol':'1.2-2'
          }}
        </td>
      </tr>
    </tbody>
  </table>
  <table>
    <thead>
      <tr>
        <th>
          <h3>One-Time Payment Method Installment Fees</h3>
        </th>
        <th></th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Checking or Savings Account</td>
        <td>
          {{ payment?.autopayInstallmentFee?.otpFees?.eft | currency: 'USD':'symbol':'1.2-2' }}
        </td>
      </tr>
      <tr>
        <td>Debit Card</td>
        <td>
          {{
            payment?.autopayInstallmentFee?.otpFees?.pciDebitCard | currency: 'USD':'symbol':'1.2-2'
          }}
        </td>
      </tr>
      <tr>
        <td>Credit Card</td>
        <td>
          {{
            payment?.autopayInstallmentFee?.otpFees?.pciCreditCard
              | currency: 'USD':'symbol':'1.2-2'
          }}
        </td>
      </tr>
    </tbody>
  </table>
</div>
