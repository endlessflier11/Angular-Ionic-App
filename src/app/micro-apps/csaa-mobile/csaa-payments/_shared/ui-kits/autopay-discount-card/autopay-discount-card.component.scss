@import '../../../../_scss/variables';

ion-item {
  --detail-icon-opacity: 1;
  --detail-icon-color: var(--ion-color-primary);
}
ion-label {
  h3 {
    margin-top: 0;
    color: var(--ion-color-primary);
  }
}
.discount {
  padding: 0 calc(var(--ion-padding) * 2);
  height: 0;
  overflow: hidden;

  &.active {
    height: auto;
  }

  table {
    width: 100%;
    border-collapse: separate;
    border-spacing: 0 10px;

    thead th {
      text-align: left;
      h3 {
        margin: 0;
      }
    }

    td,
    th {
      //padding: 8px;
      text-align: left;

      &:last-child {
        text-align: right;
      }
    }
  }
}
