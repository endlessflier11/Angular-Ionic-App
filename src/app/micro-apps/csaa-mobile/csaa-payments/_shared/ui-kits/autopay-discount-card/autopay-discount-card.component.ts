import { Component, Input, OnInit } from '@angular/core';
import { AnalyticsService, CsaaTheme, RouterService } from '../../../../_core/services';
import { noop } from 'rxjs';
import {
  Category,
  EventName,
  EventType,
  PaymentAccount,
  Policy,
  UpcomingPayment,
} from '../../../../_core/interfaces';
import { ConfigState } from '../../../../_core/store/states/config.state';
import { Store } from '@ngxs/store';

@Component({
  selector: 'csaa-autopay-discount-card',
  templateUrl: './autopay-discount-card.component.html',
  styleUrls: ['./autopay-discount-card.component.scss'],
})
export class AutopayDiscountCardComponent implements OnInit {
  @Input() policy: Policy;
  @Input() navigate: boolean;
  @Input() payment: UpcomingPayment;
  @Input() paymentAccount: PaymentAccount;
  currentTheme: CsaaTheme;
  isOpen = false;

  public get policyNumber(): string {
    return this.policy?.number;
  }

  public get discountDataDiff(): number {
    if (!this.payment || !this.payment?.autopayInstallmentFee) {
      return null;
    }
    const { autoPayFees, otpFees } = this.payment?.autopayInstallmentFee;
    const lowestAutopayFee = Math.min(
      autoPayFees.eft,
      autoPayFees.pciDebitCard,
      autoPayFees.pciCreditCard
    );
    const highestOTPFee = Math.max(otpFees.eft, otpFees.pciDebitCard, otpFees.pciCreditCard);
    return highestOTPFee - lowestAutopayFee;
  }

  constructor(
    private readonly store: Store,
    private readonly routerService: RouterService,
    private readonly analyticsService: AnalyticsService
  ) {}

  ngOnInit(): void {
    this.currentTheme = this.store.selectSnapshot(ConfigState.theme);
  }

  public clickHandler(): void {
    if (!this.navigate) {
      this.toggle();
    } else {
      this.analyticsService.trackEvent(EventName.SAVE_WITH_AUTOPAY_SELECTED, Category.payments, {
        event_type: EventType.LINK_ACCESSED,
        ...AnalyticsService.mapPolicy(this.policy),
        // eslint-disable-next-line @typescript-eslint/naming-convention
        potential_savings: this.discountDataDiff,
      });
      this.routerService
        .navigateForward('csaa.payment.autopay.settings', { policyNumber: this.policyNumber })
        .then(noop);
    }
  }

  public toggle(): void {
    this.isOpen = !this.isOpen;
    if (this.isOpen) {
      this.analyticsService.trackEvent(
        EventName.AUTOPAY_INSTALLMENT_TABLE_EXPANDED,
        Category.payments,
        {
          event_type: EventType.UX_MODIFIED,
          ...AnalyticsService.mapPolicy(this.policy),
          // eslint-disable-next-line @typescript-eslint/naming-convention
          potential_savings: this.discountDataDiff,
        }
      );
    }
  }
}
