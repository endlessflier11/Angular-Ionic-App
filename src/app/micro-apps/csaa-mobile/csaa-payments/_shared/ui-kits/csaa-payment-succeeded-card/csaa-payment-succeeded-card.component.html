<ion-card class="csaa-card">
  <ion-card-header>
    <img [src]="'/assets/csaa-mobile/vectors/success.svg'" />
    <ion-card-title>Payment Successful</ion-card-title>
  </ion-card-header>
  <ion-card-content>
    <ion-grid>
      <h2>{{ payment.policyNumber }}</h2>
      <ion-row class="item">
        <ion-col>
          <div class="policy-meta-content">
            <h3>
              Paid on
              {{ result?.date | date: 'mediumDate' }}
            </h3>
            <span class="subtitle">
              <csaa-policy-subtitles [policySubtitle]="payment.subtitle"></csaa-policy-subtitles>
            </span>
          </div>
        </ion-col>
        <ion-col size="auto">
          <div class="policy-meta-icon">
            <csaa-theme-icon slot="end" [type]="payment.policyType" [size]="40"></csaa-theme-icon>
          </div>
        </ion-col>
      </ion-row>
    </ion-grid>

    <ion-list>
      <ion-item class="csaa-payment-amount-list-item ion-no-padding">
        <ion-label>Amount paid </ion-label>
        <span slot="end">{{ result?.amount | currency: 'USD':'symbol':'1.2-2' }}</span>
      </ion-item>
      <ion-item lines="none" class="csaa-payment-method-list-item ion-no-padding">
        <ion-avatar slot="start" *ngIf="result?.paymentAccount">
          <span>
            <img
              *ngIf="result?.paymentAccount?.card?.type"
              [src]="'/assets/csaa-mobile/vectors/' + result?.paymentAccount.card.type + '.svg'"
            />
            <img
              *ngIf="result?.paymentAccount?.account"
              [src]="'/assets/csaa-mobile/vectors/bank.svg'"
            />
          </span>
        </ion-avatar>
        <ion-label
          [ngClass]="{
            'card-expired': result?.paymentAccount?.card && result?.paymentAccount?.card.expired
          }"
          class="ion-text-nowrap"
        >
          <span>{{ result?.paymentAccount ? result?.paymentAccount.shortName : 'Paid With' }}</span>
        </ion-label>
        <span
          slot="end"
          *ngIf="result?.paymentAccount?.card && result?.paymentAccount?.card?.isDebitCard !== true"
        >
          {{ 'Credit' }}
        </span>
        <span slot="end" *ngIf="result?.paymentAccount?.card?.isDebitCard">{{ 'Debit' }}</span>
        <span slot="end" *ngIf="result?.paymentAccount?.account">{{
          result?.paymentAccount?.account?.type | titlecase
        }}</span>
      </ion-item>
      <ion-item class="ion-no-padding" [class.ion-margin-bottom]="!payment?.autopayEnrollment">
        <ion-label>Receipt number </ion-label>
        <span slot="end">{{ result?.receiptNumber }}</span>
      </ion-item>
      <ng-container *ngIf="!payment?.autopayEnrollment">
        <csaa-autopay-discount-card
          [policy]="policy"
          [payment]="payment"
          [navigate]="true"
          [paymentAccount]="result?.paymentAccount"
        ></csaa-autopay-discount-card>
      </ng-container>
    </ion-list>
  </ion-card-content>
</ion-card>
