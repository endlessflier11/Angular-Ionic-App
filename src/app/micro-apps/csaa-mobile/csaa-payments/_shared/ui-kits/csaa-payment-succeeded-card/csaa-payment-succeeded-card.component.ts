import { Component, Input } from '@angular/core';
import { Policy, PolicyPaymentResult, UpcomingPayment } from '../../../../_core/interfaces';

@Component({
  selector: 'csaa-payment-succeeded-card',
  templateUrl: 'csaa-payment-succeeded-card.component.html',
  styleUrls: ['csaa-payment-succeeded-card.scss'],
})
export class CsaaPaymentSucceededCardComponent {
  @Input() policy: Policy;
  @Input() payment: UpcomingPayment;
  @Input() loading = false;
  @Input() result: PolicyPaymentResult;
}
