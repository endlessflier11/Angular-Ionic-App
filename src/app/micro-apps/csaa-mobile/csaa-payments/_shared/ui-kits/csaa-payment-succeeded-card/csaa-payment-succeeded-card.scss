@import '../../../../_scss/variables';

ion-card {
  ion-card-header {
    text-align: center;
    > img {
      margin-bottom: 14px;
    }
  }

  ion-card-content {
    h2 {
      font-size: 24px;
      margin-top: 30px;
      margin-bottom: 5px;
    }
    .policy-meta-content {
      h3 {
        color: var(--csaa-color-success);
        font-weight: $csaa-active-font-weight;
      }
    }
    ion-list {
      ion-item {
        font-size: 16px;
        > span {
          color: $csaa-lighter-color;
        }
        &:first-child {
          color: var(--csaa-color-success);
          > span {
            color: var(--csaa-color-success);
          }
        }

        &.csaa-payment-method-list-item {
          border-bottom: $csaa-list-item-border;
        }
      }
    }
  }
}
