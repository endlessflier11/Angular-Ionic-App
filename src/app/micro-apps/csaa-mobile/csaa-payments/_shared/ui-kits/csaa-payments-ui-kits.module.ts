import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaPaymentHistoryCardComponent } from './csaa-payment-history-card/csaa-payment-history-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentHistorySelectCardComponent } from './csaa-payment-history-select-card/csaa-payment-history-select-card.component';
import { CsaaPaymentSucceededCardComponent } from './csaa-payment-succeeded-card/csaa-payment-succeeded-card.component';
import { CsaaMakePaymentCardComponent } from './csaa-make-payment-card/csaa-make-payment-card.component';
import { CsaaPayAllPoliciesCardComponent } from './csaa-pay-all-policies-card/csaa-pay-all-policies-card.component';
import { PaymentTermsConditionsComponent } from './payment-terms-conditions/payment-terms-conditions.component';
import { AutopayTermsConditionsComponent } from './autopay-terms-conditions/autopay-terms-conditions.component';
import { CsaaBillsCardComponent } from './csaa-bills-card/csaa-bills-card.component';
import { AutopayDiscountCardComponent } from './autopay-discount-card/autopay-discount-card.component';

const SHARED_COMPONENTS = [
  CsaaPaymentHistoryCardComponent,
  CsaaPaymentHistorySelectCardComponent,
  CsaaMakePaymentCardComponent,
  CsaaPaymentSucceededCardComponent,
  CsaaPayAllPoliciesCardComponent,
  PaymentTermsConditionsComponent,
  AutopayTermsConditionsComponent,
  CsaaBillsCardComponent,
  AutopayDiscountCardComponent,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule, IonicModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaPaymentsUiKitsModule {}
