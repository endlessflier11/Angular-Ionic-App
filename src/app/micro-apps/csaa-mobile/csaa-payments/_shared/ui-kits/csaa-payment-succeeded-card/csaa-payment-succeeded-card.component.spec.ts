import { CommonModule } from '@angular/common';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { IonicModule } from '@ionic/angular';
import { PageTestingModule } from '../../../../../../../testing';
import { PaymentType } from '../../../../_core/interfaces/payment.interface';
import { PolicyType } from '../../../../_core/interfaces/policy.interface';
import DateHelper from '../../../../_core/shared/date.helper';
import { UiKitsModule } from '../../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentSucceededCardComponent } from './csaa-payment-succeeded-card.component';
import { AutopayDiscountCardComponent } from '../autopay-discount-card/autopay-discount-card.component';

describe('CsaaPaymentSelectModal', () => {
  let component: CsaaPaymentSucceededCardComponent;
  let fixture: ComponentFixture<CsaaPaymentSucceededCardComponent>;

  beforeEach(async () => {
    TestBed.configureTestingModule({
      declarations: [CsaaPaymentSucceededCardComponent, AutopayDiscountCardComponent],
      imports: [
        CommonModule,
        IonicModule,
        UiKitsModule,
        PageTestingModule.withConfig({ providesStorage: true, providesAnalytics: true }),
      ],
      providers: [],
    });

    fixture = TestBed.createComponent(CsaaPaymentSucceededCardComponent);
    component = fixture.componentInstance;

    component.payment = component.payment = {
      vehicles: ['Toyota Prius'],
      subtitle: 'test',
      autoPay: false,
      minimumAmount: 20,
      remainingPremium: 10000,
      otherAmount: 1023.45,
      type: PaymentType.other,
      nextDueDate: DateHelper.toDate('2040-08-19'),
      nextAmount: 30,
      allPolicies: false,
      policyNumber: 'AZSS294857564',
      policyType: PolicyType.Auto,
      dueDate: DateHelper.toDate('2040-08-19'),
      isPastDue: false,
      isPaymentDue: true,
      amount: 148.1,
      policyRiskState: 'test',
      autopayEnrollment: null,
    };
    component.result = {
      result: true,
      receiptNumber: '1668324',
      paymentAccount: {
        paymentAccountToken: '0AC5D0B0-5383-BBB2-6EF6-0E7C35B81E7D',
        isPreferred: true,
        shortName: 'Tt 5454 5454',
        account: null,
        card: {
          type: 'mastercard',
          isDebitCard: false,
          last4digits: '5454',
          holderName: 'TEST USER',
          printedName: 'TEST USER',
          expirationDate: '2025-12-01',
          zipCode: '22911',
          expired: false,
        },
      },
      amount: 1023.45,
      date: '2021-04-09',
    };

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshots', () => {
    expect(fixture).toMatchSnapshot();
  });
});
