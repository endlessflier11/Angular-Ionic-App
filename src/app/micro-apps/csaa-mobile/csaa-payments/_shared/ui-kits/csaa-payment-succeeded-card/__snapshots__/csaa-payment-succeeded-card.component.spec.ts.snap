// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsaaPaymentSelectModal should match snapshots 1`] = `
<csaa-payment-succeeded-card
  __ngContext__={[Function LRootView]}
  loading="false"
  payment={[Function Object]}
  result={[Function Object]}
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <img
        src="/assets/csaa-mobile/vectors/success.svg"
      />
      <ion-card-title>
        Payment Successful
      </ion-card-title>
    </ion-card-header>
    <ion-card-content>
      <ion-grid>
        <h2>
          AZSS294857564
        </h2>
        <ion-row
          class="item"
        >
          <ion-col>
            <div
              class="policy-meta-content"
            >
              <h3>
                 Paid on Apr 9, 2021 
              </h3>
              <span
                class="subtitle"
              >
                <csaa-policy-subtitles>
                  <span
                    class="subtitle"
                  >
                    test
                  </span>
                </csaa-policy-subtitles>
              </span>
            </div>
          </ion-col>
          <ion-col
            size="auto"
          >
            <div
              class="policy-meta-icon"
            >
              <csaa-theme-icon
                slot="end"
              >
                <img
                  src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                  style="width: 40px; height: 40px;"
                />
              </csaa-theme-icon>
            </div>
          </ion-col>
        </ion-row>
      </ion-grid>
      <ion-list>
        <ion-item
          class="csaa-payment-amount-list-item ion-no-padding"
        >
          <ion-label>
            Amount paid 
          </ion-label>
          <span
            slot="end"
          >
            $1,023.45
          </span>
        </ion-item>
        <ion-item
          class="csaa-payment-method-list-item ion-no-padding"
          lines="none"
        >
          <ion-avatar
            slot="start"
          >
            <span>
              <img
                src="/assets/csaa-mobile/vectors/mastercard.svg"
              />
            </span>
          </ion-avatar>
          <ion-label
            class="ion-text-nowrap"
          >
            <span>
              Tt 5454 5454
            </span>
          </ion-label>
          <span
            slot="end"
          >
             Credit 
          </span>
        </ion-item>
        <ion-item
          class="ion-no-padding ion-margin-bottom"
        >
          <ion-label>
            Receipt number 
          </ion-label>
          <span
            slot="end"
          >
            1668324
          </span>
        </ion-item>
        <csaa-autopay-discount-card>
          <ion-item
            class="ion-no-padding"
            detail=""
            lines="none"
          >
            <ion-label>
              <h3>
                Save money with AutoPay
              </h3>
              <ion-text
                color="dark"
              >
                <p
                  class="ion-text-wrap"
                >
                   Save up to  per payment with 
                  <br />
                  AutoPay. Tap to view fees 
                </p>
              </ion-text>
            </ion-label>
          </ion-item>
          <div
            class="discount"
          >
            <table>
              <thead>
                <tr>
                  <th>
                    <h3>
                      Auto Payment Method Installment Fees
                    </h3>
                  </th>
                  <th />
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    Checking or Savings Account
                  </td>
                  <td>
                  </td>
                </tr>
                <tr>
                  <td>
                    Debit Card
                  </td>
                  <td>
                  </td>
                </tr>
                <tr>
                  <td>
                    Credit Card
                  </td>
                  <td>
                  </td>
                </tr>
              </tbody>
            </table>
            <table>
              <thead>
                <tr>
                  <th>
                    <h3>
                      One-Time Payment Method Installment Fees
                    </h3>
                  </th>
                  <th />
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    Checking or Savings Account
                  </td>
                  <td>
                  </td>
                </tr>
                <tr>
                  <td>
                    Debit Card
                  </td>
                  <td>
                  </td>
                </tr>
                <tr>
                  <td>
                    Credit Card
                  </td>
                  <td>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </csaa-autopay-discount-card>
      </ion-list>
    </ion-card-content>
  </ion-card>
</csaa-payment-succeeded-card>
`;
